be/user.go

```diff
@@ -63,5 +63,10 @@ func UpdateUser(c *gin.Context) {
 			return
 		}
 	}
+
+	if user.Referral != "" {
+		// TODO: Make the call async after the db connection is handled by the db controller
+		cleanupReferral(db, user.Referral)
```

Comment:
? So, a referral code is used only once? That's not our original design.
Also what happens when updating user?

Reply:
Good catch, the index by ref_value shouldn't be unique. I changed that.

And yes, the refcode can be used once by the user i.e. one device at certain IP address. The ref code can be used multiple times by different devices and IP addresses.

Reply:
Good.